tests/test_api/test_items.py

```diff
@@ -228,15 +228,13 @@ def test_get_items(self, test_client, item):
 class TestCategoryItemListAllAPI:
 
     def test_get_category_items_missing_category(self, test_client):
-        r = test_client.get('/api/categories/nonexistent_category/items/all')
+        r = test_client.get('/api/categories/100/items/all')
         check_valid_header_type(r.headers)
-        assert r.status_code == 200
-
-        data = json.loads(r.data)
-        assert len(data['items']) == 0
+        assert r.status_code == 404
 
     def test_get_category_items_present(self, test_client, itemsWithCategory):
-        r = test_client.get('/api/categories/' + itemsWithCategory[0].category_1 + '/items/all')
+        category = itemsWithCategory[0].cats.first()
```

Comment:
.cats is the actual association. .categories is just the association proxy.

Association proxies just work as a convenience for when you're constructing/modifying models. To actually pull out the category models, rather than just their unicode representation (as strings), we need to use cats.